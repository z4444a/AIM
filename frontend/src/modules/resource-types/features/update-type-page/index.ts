import { UpdateTypePage } from './update-type-page';
import { connect } from 'react-redux';
import { updateResourceType } from '../../../../redux/actions/resource-type-actions';
import { TypeUpdateModel } from '../../../../model/update/type-update-model';
import { DeepPartial } from 'redux';
import { fetchType } from '../../../../redux/actions/type-action';
import { RootState } from 'typesafe-actions';
import {
  findParameterValue,
  resetRemoveParameterResponse,
} from '../../../../redux/actions/resource-type-page-actions';

export const mapDispatchToProps = (dispatch: Function) => ({
  fetchType: (id: number) => dispatch(fetchType.request(id)),
  updateType: (model: DeepPartial<TypeUpdateModel>) => dispatch(updateResourceType.request(model)),
  findParamValue: (id: number) => dispatch(findParameterValue.request(id)),
  resetRemoveParameterResponse: () => dispatch(resetRemoveParameterResponse()),
});
export const mapStateToProps = (state: RootState) => ({
  oldModel: state.resourceTypePage.previousModelState,
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(UpdateTypePage);
