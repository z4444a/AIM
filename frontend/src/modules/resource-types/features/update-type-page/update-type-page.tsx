import React from 'react';
import { mapDispatchToProps, mapStateToProps } from './index';
import { RouteComponentProps } from 'react-router';
import ResourceTypePage from '../resource-type-page/index';

export type InternalProps = ReturnType<typeof mapDispatchToProps> &
  ReturnType<typeof mapStateToProps> &
  RouteComponentProps<{ id: string }>;

export class UpdateTypePage extends React.PureComponent<InternalProps> {
  public componentDidMount(): void {
    const { match, fetchType } = this.props;
    const id = parseInt(match.params.id, 10);
    fetchType(id);
  }
  public render(): React.ReactNode {
    const { updateType, oldModel, findParamValue, resetRemoveParameterResponse } = this.props;
    return (
      <ResourceTypePage
        showIdentifiers
        onSubmit={updateType}
        oldModel={oldModel}
        findParamValue={findParamValue}
        resetRemoveParameterResponse={resetRemoveParameterResponse}
      />
    );
  }
}
