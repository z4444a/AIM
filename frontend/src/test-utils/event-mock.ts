export const getMockChangeEvent = (value: string) => {
  return {
    target: {
      value: value,
    },
  };
};

export const getMockKeyPressEvent = (key: string) => {
  return {
    key: key,
    preventDefault: jest.fn().mockName('preventDefault'),
  };
};
