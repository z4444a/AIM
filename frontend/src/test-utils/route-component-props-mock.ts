import { RouteComponentProps } from 'react-router';
import { Href } from 'history';

export function getMockRouterProps<P>(data: P): RouteComponentProps {
  const location = {
    hash: '',
    key: '',
    pathname: '',
    search: '',
    state: {},
  };

  const props: RouteComponentProps<P> = {
    match: {
      isExact: true,
      params: data,
      path: '',
      url: '',
    },
    location: location,
    history: {
      length: 2,
      action: 'POP',
      location: location,
      push: () => {},
      replace: () => {},
      go: () => {},
      goBack: () => {},
      goForward: () => {},
      block: () => () => {},
      createHref: () => {
        const temp: Href = '';
        return temp;
      },
      listen: () => () => {},
    },
    staticContext: {},
  };

  return props;
}
