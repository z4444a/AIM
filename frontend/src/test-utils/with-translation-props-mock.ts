import { WithTranslation } from 'react-i18next';

export function getMockWithTranslationProps(): WithTranslation {
  // eslint-disable-next-line
  const t = ((key: string) => `${key}_i18n`) as any;
  // eslint-disable-next-line
  const i18n = {} as any;
  const props: WithTranslation = {
    tReady: true,
    t,
    i18n,
  };

  return props;
}
