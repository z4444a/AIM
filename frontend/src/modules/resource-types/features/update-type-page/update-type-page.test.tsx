import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { InternalProps, UpdateTypePage } from './update-type-page';
import { getMockRouterProps } from '../../../../test-utils/route-component-props-mock';

describe('UpdateTypePage', () => {
  let wrapper: ShallowWrapper;
  beforeEach(() => {
    const props: InternalProps = {
      ...getMockRouterProps<{ id: string }>({ id: '1' }),
      fetchType: id => null,
      updateType: model => null,
    };

    wrapper = shallow(<UpdateTypePage {...props} />);
  });

  it('should render correctly', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
