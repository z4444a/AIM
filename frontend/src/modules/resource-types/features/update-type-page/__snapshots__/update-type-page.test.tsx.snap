// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpdateTypePage should render correctly 1`] = `
<Connect(WithStyles(ResourceTypePage))
  onSubmit={[Function]}
  showIdentifiers={true}
/>
`;
